File: src/gen5/Gen5TimerMode.cpp
//
// Created by Dylan Meadows on 2020-03-19.
//

#include "Gen5TimerMode.h"

namespace EonTimer::Gen5 {
    const std::vector<Gen5TimerMode> &getGen5TimerModes() {
        static const std::vector<Gen5TimerMode> values{STANDARD, C_GEAR, ENTRALINK, ENTRALINK_PLUS};
        return values;
    }

    const QString &getName(const Gen5TimerMode mode) {
        static const QString names[COUNT] = {"Standard", "C-Gear", "Entralink", "Entralink+"};
        return names[mode];
    }
}  // namespace EonTimer

File: src/main/kotlin/io/eontimer/gen5/Gen5TimerMode.kt
package io.eontimer.gen5

import io.eontimer.util.javafx.Choice

enum class Gen5TimerMode(
    override val displayName: String
) : Choice {
    STANDARD("Standard"),
    C_GEAR("C-Gear"),
    ENTRALINK("Entralink"),
    ENHANCED_ENTRALINK("Entralink+")
}